src/server/saved_objects/migrations/README.md

```diff
@@ -0,0 +1,115 @@
+# Migrations
+
+This is the system that manages the versioning of saved object indices. It manages changes to the index, including mappings, index templates, and changes to the shape of documents from one version of Kibana to another.
+
+## Outstanding questions
+
+- Should we migrate all docs in the index between minor versions?
+  - @epixa does not want to migrate the index on minor versions
+  - @clintongormley [seems to think we should](https://github.com/elastic/kibana/issues/15100#issuecomment-398041512)
```

Comment:
If you don't how would you manage this situation:

* core is upgraded to 7.2
* user reinstalls plugin
* plugins needs to upgrade docs from 6.5

Reply:
One idea had been that when migrating between major versions, we would drop docs which were not supported. But yes. I think that things are across the board simpler if we just always migrate the index.